### PR TITLE
feat(ui5-side-navigation): add external link icon

### PR DESCRIPTION
**Pull request by kskondov**
if the navigation item is an external link, show the external link indicator

### CODE REVIEW

**Review by LidiyaGeorgieva — COMMENTED** [review]
expand/collapse arrows are missing in this sample:
https://sap.github.io/ui5-webcomponents/nightly/playground/fiori/pages/SideNavigationOnly/?sap-ui-theme=sap_horizon#events

File: packages/fiori/src/SideNavigation.hbs [hunk 1]
```diff
@@ -136,10 +148,15 @@
...
 				></ui5-icon>
 				{{#if items.length}}
-					<ui5-icon class="ui5-sn-item-toggle-icon"
-							  name="{{_toggleIconName}}"
-							  @click="{{_onToggleClick}}"
-					></ui5-icon>
+				<ui5-icon class="ui5-sn-item-toggle-icon"
+						  name="{{_toggleIconName}}"
+						  @click="{{_onToggleClick}}"
+				></ui5-icon>
```
Comment (LidiyaGeorgieva) [inline packages/fiori/src/SideNavigation.hbs:154]: please fix indentation

File: packages/fiori/src/SideNavigation.hbs [hunk 2]
```diff
@@ -136,10 +148,15 @@
...
+				{{/if}}
+				{{#if _isExternalLink}}
+				<ui5-icon class="ui5-sn-item-external-link-icon"
+						  name="arrow-right"
+				></ui5-icon>
```
Comment (LidiyaGeorgieva) [inline packages/fiori/src/SideNavigation.hbs:159]: please fix indentation

File: packages/fiori/src/SideNavigation.hbs [hunk 3]
```diff
@@ -200,6 +222,11 @@
 								<ui5-icon class="ui5-sn-item-selection-icon"
 										  name="circle-task-2"
 								></ui5-icon>
+								{{#if _isExternalLink}}
+								<ui5-icon class="ui5-sn-item-external-link-icon"
```
Comment (LidiyaGeorgieva) [inline packages/fiori/src/SideNavigation.hbs:226]: please fix indentation

File: packages/main/src/NavigationMenu.hbs [hunk 4]
```diff
@@ -97,6 +97,11 @@
                                 >
                                 </div>
                             {{/if}}
+							{{#if this.item.target}}
+							<ui5-icon class="ui5-sn-item-external-link-icon"
```
Comment (LidiyaGeorgieva) [inline packages/main/src/NavigationMenu.hbs:101]: please fix indentation

**Review by LidiyaGeorgieva — COMMENTED** [review]
Please add more samples:
- in the Story books
- in the fixed part of the Side Navigation
- maybe example with disabled item

Please add tests.

Please check again if the color of the arrow is correct:
https://wiki.one.int.sap/wiki/display/CPDesign/Vertical+Navigation
image
https://wiki.one.int.sap/wiki/pages/viewpage.action?pageId=2650676561
image
currently is "#1d2d3e;" for Horizon

File: packages/fiori/src/SideNavigation.hbs [hunk 5]
```diff
@@ -73,6 +73,7 @@
 		   title="{{_tooltip}}"
 		   href="{{_href}}"
 		   target="{{_target}}"
+		   externalLink="{{_isExternalLink}}"
```
Comment (TeodorTaushanov) [inline packages/fiori/src/SideNavigation.hbs:76]: not needed

File: packages/fiori/src/SideNavigation.hbs [hunk 6]
```diff
@@ -127,20 +138,26 @@
 			   aria-owns="{{_groupId}}"
 			   href="{{_href}}"
 			   target="{{_target}}"
+			   externalLink="{{_isExternalLink}}"
```
Comment (TeodorTaushanov) [inline packages/fiori/src/SideNavigation.hbs:141]: not needed

File: packages/fiori/src/SideNavigationItemBase.ts [hunk 7]
```diff
@@ -115,6 +115,9 @@ class SideNavigationItemBase extends UI5Element implements ITabbable {
 	@property({ defaultValue: "-1", noAttribute: true })
 	_tabIndex!: string;
 
+	@property({ type: Boolean })
+	_externalLink!: boolean;
```
Comment (TeodorTaushanov) [inline packages/fiori/src/SideNavigationItemBase.ts:119]: this is not used anywhere

**Review by TeodorTaushanov — CHANGES_REQUESTED** [review]
External links in the overflow menu are not opened by keyboard Enter and Space key

**Review by LidiyaGeorgieva — COMMENTED** [review]
In SideNavigationOnly.html test page the arrow for expand/collapse is grey now. 
image
This is not correct.

**Review by LidiyaGeorgieva — COMMENTED** [review]
+1

**Review by TeodorTaushanov — APPROVED** [review]

**Pull request merged**